### DISCUSSION TITLE
@vanilla-extract plugin not working

### DISCUSSION THREAD

**Discussion by mc-petry** [wevm/vocs#57] [Q&A]
Plugin [@vanilla-extract/vite-plugin](https://www.npmjs.com/package/@vanilla-extract/vite-plugin) is not working:
I've got an errors like: `Error: Identifier function returned invalid indentifier: "vocs_block.css.ts"`

Tried latest plugin version: [3.9.4](https://www.npmjs.com/package/@vanilla-extract/vite-plugin/v/3.9.4)
And future version [0.0.0-vite-plugin-new-20240125071352](https://www.npmjs.com/package/@vanilla-extract/vite-plugin/v/0.0.0-vite-plugin-new-20240125071352)

This plugin works well in pure `vite`, also it works well in astro.build.
Plugin transforms `*.css.ts` to `css`
[Plugin code](https://github.com/vanilla-extract-css/vanilla-extract/blob/master/packages/vite-plugin/src/index.ts)
[Example](https://codesandbox.io/p/devbox/vocs-vanilla-extract-cgnk7d?file=%2Fpackage.json%3A12%2C38&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrvvghth00073b69z7rb0af5%2522%252C%2522sizes%2522%253A%255B63.25481799%252C36.74518201%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrvvghtg00023b69mmbyn3ha%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrvvghtg00043b69hvrusasc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrvvghtg00063b69q07frnyq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B53.46028164216756%252C46.53971835783244%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrvvghtg00023b69mmbyn3ha%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00023b69mmbyn3ha%2522%252C%2522activeTabId%2522%253A%2522cls0jgc8r00023b6bmh16f90t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00013b69nrpgj6jy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clrvwuoeg02593b69qegnm9g3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clrvxekfg05v03b69w7blp561%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A17%252C%2522startColumn%2522%253A5%252C%2522endLineNumber%2522%253A17%252C%2522endColumn%2522%253A5%257D%255D%252C%2522filepath%2522%253A%2522%252Fvocs.config.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cls0jgc8r00023b6bmh16f90t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A12%252C%2522startColumn%2522%253A38%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A38%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clrvvghtg00063b69q07frnyq%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00063b69q07frnyq%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00053b69i6ui1yw1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522pnpm%2520dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clrvwu7s0023n3b693vjowh0n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522id%2522%253A%2522cls0jgxvn002v3b6bajl7v2ll%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrvvghtg00053b69i6ui1yw1%2522%257D%252C%2522clrvvghtg00043b69hvrusasc%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00043b69hvrusasc%2522%252C%2522activeTabId%2522%253A%2522clrvx5ysb04dp3b69sray0jzz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00033b69mopmhi83%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Ftypescript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522clrvwqf9u01mp3b697qrxwv5y%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522pnpm%2520dev%2522%252C%2522id%2522%253A%2522clrvx5ysb04dp3b69sray0jzz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by jxom** [comment]
Can you please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Reply by mc-petry** [reply]
Sure: [Example](https://codesandbox.io/p/devbox/vocs-vanilla-extract-cgnk7d?file=%2Fpackage.json%3A12%2C38&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrvvghth00073b69z7rb0af5%2522%252C%2522sizes%2522%253A%255B63.25481799%252C36.74518201%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrvvghtg00023b69mmbyn3ha%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrvvghtg00043b69hvrusasc%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrvvghtg00063b69q07frnyq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B53.46028164216756%252C46.53971835783244%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrvvghtg00023b69mmbyn3ha%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00023b69mmbyn3ha%2522%252C%2522activeTabId%2522%253A%2522cls0jgc8r00023b6bmh16f90t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00013b69nrpgj6jy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clrvwuoeg02593b69qegnm9g3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clrvxekfg05v03b69w7blp561%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A17%252C%2522startColumn%2522%253A5%252C%2522endLineNumber%2522%253A17%252C%2522endColumn%2522%253A5%257D%255D%252C%2522filepath%2522%253A%2522%252Fvocs.config.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cls0jgc8r00023b6bmh16f90t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A12%252C%2522startColumn%2522%253A38%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A38%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clrvvghtg00063b69q07frnyq%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00063b69q07frnyq%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00053b69i6ui1yw1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522pnpm%2520dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clrvwu7s0023n3b693vjowh0n%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522id%2522%253A%2522cls0jgxvn002v3b6bajl7v2ll%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrvvghtg00053b69i6ui1yw1%2522%257D%252C%2522clrvvghtg00043b69hvrusasc%2522%253A%257B%2522id%2522%253A%2522clrvvghtg00043b69hvrusasc%2522%252C%2522activeTabId%2522%253A%2522clrvx5ysb04dp3b69sray0jzz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrvvghtg00033b69mopmhi83%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Ftypescript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522clrvwqf9u01mp3b697qrxwv5y%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522pnpm%2520dev%2522%252C%2522id%2522%253A%2522clrvx5ysb04dp3b69sray0jzz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
